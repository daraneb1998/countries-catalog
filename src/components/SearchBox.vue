<script setup lang="ts">
import { useDebounceFn } from "@vueuse/core";

import IconSearch from "@/components/icons/IconSearch.vue";
import { useSearchStore } from "@/stores/search";
import { CONSTANTS } from "@/constants";

const searchStore = useSearchStore();

const handleChange = (event: Event) => {
  const target = event.target as HTMLInputElement;
  searchStore.setSearchKeyword(target.value);
};
const handleDebouncedSearch = useDebounceFn((event: Event) => {
  searchStore.updateCurrentPageNumber(CONSTANTS.INITIAL_PAGE_NUMBER);
  handleChange(event);
}, CONSTANTS.DEBOUNCED_SEARCH_DURATION);
</script>

<template>
  <div class="sticky flex flex-row md:w-1/3 m-auto h-12 md:my-5 w-[calc(100vw-61px)]">
    <div
      class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
    >
      <IconSearch />
    </div>
    <input
      id="default-search"
      type="text"
      class="block p-4 pl-10 w-full rounded-sm border border-blue-500 focus:outline-none focus:border-blue-700 bg-sky-50"
      placeholder="COUNTRY NAME"
      :value="searchStore.searchKeyword"
      @input="handleDebouncedSearch"
    />
  </div>
</template>
