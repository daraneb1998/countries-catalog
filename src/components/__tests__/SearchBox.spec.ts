import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import SearchBox from "../SearchBox.vue";

describe("SearchBox", () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it("renders properly", () => {
    const wrapper = mount(SearchBox);
    expect(wrapper.text()).toMatchSnapshot();
  });
});
