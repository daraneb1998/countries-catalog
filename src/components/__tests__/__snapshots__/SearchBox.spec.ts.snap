// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > renders properly 1`] = `"Page 1 (25) of 0 PREVIOUS  NEXT"`;

exports[`SearchBox > renders properly 1`] = `""`;
